Code-4-Community/speak-for-the-trees-frontend
# Edit stewardship activity functionality

## Checklist

- [x] 1. Run `npm run check`
- [x] 2. Run `npm run test`
- [x] 3. Change ClickUp ticket status to "In Review"
- [x] 4. After making the PR, add PR link to ClickUp ticket

## Why

[ClickUp Ticket](https://app.clickup.com/t/3ucv1u9)

Give users the ability to edit existing stewardship activities (change date + actions taken). Admins can edit any activity, and volunteers can edit any they've made.



## This PR

- Refactor `CareEntry` and delete activity functionality from `TreeActivity` into a new component  
- Add edit stewardship route to router and `onFinishEditStewardship` function to perform edit
- Update `StewardshipForm` to take an optional initial date



## Screenshots



Page view of edit button:

![1](https://user-images.githubusercontent.com/68914997/214467930-72ad35e1-b9db-4e22-81b2-0bfef087524c.png)

Modal view with form:

![2](https://user-images.githubusercontent.com/68914997/214467933-1c3bb803-e634-4495-a8ad-c03ead1cf5a1.png)

After submitting changes:

![3](https://user-images.githubusercontent.com/68914997/214467991-f04ee9c7-cdbb-45ee-9f06-8d9fe3d70c13.png)

## Verification Steps

- As admin, activity owner, verified edit button appears and edits are successfully saved
- As admin, activity non-owner, verified edit button appears and edits are successfully saved
- As non-admin, activity owner, verified edit button appears and edits are successfully saved
- As non-admin, activity non-owner, verified edit button __does not__ appear
